### PR TITLE
Avoid default x64 activation at package import

### LINKED ISSUE #3: Remove the 'by default' X64 activation in the __init__
Remove https://github.com/CMBSciPol/jax-healpy/blob/069bea6b3c23444ee4223238c433eba9ecc1c91b/jax_healpy/__init__.py#L99

to allow users to select their prefererd precision

brought up by @sbiquard

### PR DESCRIPTION
**Pull request by sbiquard**
Closes #3. See below for a summary of changes.

**Behaviour**
- Drop the `jax.config.update('jax_enable_x64', True)` call from `jax_healpy/__init__.py` and instead warn about numerical precision and possible overflow.

**Precision-aware dtypes**
- `_pixel_dtype_for`: return `int32` for `nside <= 8192`, `int64` above.
- Type the `_JRLL`/`_JPLL` lookup tables as `int32` to avoid silent widening to `int64` under x64 in some functions.
- Use `float`/`int` dtypes in `_query_disc` and `_kmeans` instead of hard-coded `float64`/`int64`, so they follow the active precision.

**Tests**
- Add a session-wide autouse `default_x64` fixture (suite still runs in double precision, as most tests check accuracy against healpy) plus a parametrized `x64` fixture for precision-sensitive tests.
- Parametrize the ring/nest dtype and roundtrip/overflow tests over precision; skip cases that exceed the int32 range under 32-bit.
- Add invariant tests asserting no implicit 32->64 widening of integer or float outputs under x64.

**Docs**
- Add a README "Numerical Precision" section recommending enabling 64-bit for most uses and noting that `nside > 8192` requires it to avoid overflows.

### CODE REVIEW

File: tests/pixelfunc/test_interp.py [hunk 1]
```diff
@@ -157,7 +157,7 @@ def test_get_interp_weights_high_nside_sampling(region_name, nside):
     if nside <= 512:
         precision_threshold = 1e-25
     elif nside <= 1024:
```
Comment (ASKabalan) [inline tests/pixelfunc/test_interp.py:159]: why did you relax the tolerance?
is it not passing?
the tests are ran in x64

**Review by ASKabalan — COMMENTED** [review]
All good for me, except the tolerance increase

File: tests/pixelfunc/test_interp.py (hunk 1 above)
Comment (sbiquard) [reply tests/pixelfunc/test_interp.py:159]: I think the test was failing yes. Need to check again.

Comment (sbiquard) [reply tests/pixelfunc/test_interp.py:159]: Previous tolerance was `1e-10` and the test fails at Nside=1024 for the North Cap
```
E       AssertionError:
E                 High Cap region nside 1024: weight MSE 3.63e-07
E                 exceeds maximum precision threshold 1.00e-10
E
E       assert Array(3.62954902e-07, dtype=float64) < 1e-10
```

Comment (ASKabalan) [reply tests/pixelfunc/test_interp.py:159]: OK I think we can merge it then.
I will merge it in my branch and see if the regression persist

**Review by ASKabalan — APPROVED** [review]
Ok G2G
